e3db/client.py

```diff
@@ -482,51 +482,100 @@ def register(self, registration_token, client_name, public_key, private_key=None
             API url useful for testing against different e3db instances.
             Optional.
 
+        public_signing_key : str
+            base64urlencoded public_signing_key object obtained from
+            e3db.Client.generate_signing_keypair(). Optional. 
+
+        private_signing_key : str
+            base64urlencoded private_signing_key object obtained from
+            e3db.Client.generate_signing_keypair(). Optional. 
+
         Returns
         -------
         e3db.ClientDetails
             Contains API credentials, a unique client_id generated by the
             server, client name, and client public key.
         """
-
         url = "{0}/{1}/{2}/{3}/{4}/{5}".format(api_url, 'v1', 'account', 'e3db', 'clients', 'register')
-        if Crypto.get_mode() == 'nist':
-            payload = {
-                'token': registration_token,
-                'client': {
-                    'name': client_name,
-                    'public_key': {
-                        'p384': public_key
+        if public_signing_key is None:
+            if Crypto.get_mode() == 'nist':
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'p384': public_key
+                        }
+                    }
+                }
+            else:
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'curve25519': public_key
+                        }
                     }
                 }
-            }
         else:
-            payload = {
-                'token': registration_token,
-                'client': {
-                    'name': client_name,
-                    'public_key': {
-                        'curve25519': public_key
+            if Crypto.get_mode() == 'nist':
```

Comment:
nit: some ideas for reducing code duplication 

you can have less conditionals if you have the top level conditional be based off the crypto mode (which is a common case between this code and the code around line 501)

construct the base payload object first, then check if signing key is set, and if so update that base object with the signing key

e.g.
``` python
payload = {
'foo': bar 
}

if signing_key != "":
 payload. signing_key = signing_key
```

Reply:
Done